### PR TITLE
Fix DISSECT with empty patterns

### LINKED ISSUE #102577: ES|QL: empty patterns in DISSECT command return empty attribute name and null value
```
ROW a="b c d"
| DISSECT a "%{b} %{} %{d}"
```
returns

```
       a       |       b       |               |       d       
---------------+---------------+---------------+---------------
b c d          |b              |null           |d 
```

it should ignore the empty pattern instead, and only return `a, b, d`

### PR DESCRIPTION
**Pull request by luigidellaquila**
Fixes https://github.com/elastic/elasticsearch/issues/102577

Ignore empty DISSECT patterns, eg. `DISSECT a "%{a} %{} %{b}"` will only return two fields (`a` and `b`) and will ignore the empty pattern (discarding the corresponding values).

### CODE REVIEW

**Review by alex-spies — APPROVED** [review]
LGTM!

File: x-pack/plugin/esql/qa/testFixtures/src/main/resources/dissect.csv-spec [hunk 1]
```diff
@@ -158,3 +158,11 @@ emp_no:integer | a:keyword            | b:keyword         | c:keyword
 10005          | null                 | null              | null 
 10006          | [Principal, Senior]  | [Support, Team]   | [Engineer, Lead]
 ;
+
+emptyPattern#[skip:-8.11.99]
+ROW a="b c d"| DISSECT a "%{b} %{} %{d}";
+
+a:keyword | b:keyword | d:keyword
+b c d     | b         | d
+;
```
Comment (costin) [inline x-pack/plugin/esql/qa/testFixtures/src/main/resources/dissect.csv-spec:167]: For the sake of testing, please add some variants of this test with:
- multiple empty columns
- start/end with empty columns
- only empty columns

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java [hunk 2]
```diff
@@ -136,6 +136,7 @@ public PlanFactory visitDissectCommand(EsqlBaseParser.DissectCommandContext ctx)
                 }
                 List<Attribute> keys = parser.outputKeys()
                     .stream()
+                    .filter(x -> x.isEmpty() == false)
                     .map(x -> new ReferenceAttribute(src, x, DataTypes.KEYWORD))
                     .map(Attribute.class::cast)
                     .toList();
```
Comment (costin) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java:142]: It's really a preference thing - I find the code above more readable and less idiosyncratic:
```
List<Attribute> keys = new ArrayList<>();
for (var x : parser.outputKeys()) {
    if (x.isEmpty() == false) {
         keys.add(new ReferenceAttribute(src, x, DataTypes.KEYWORD));
    }
}
```

**Review by costin — APPROVED** [review]
LGTM

**Pull request merged**